pkg/volume/volume.go

```diff
@@ -94,11 +126,22 @@ func (v *VolumeOperation) Create() (*v1alpha1.CASVolume, error) {
 	// extract the cas volume config from pvc
 	casConfigPVC := pvc.Annotations[string(v1alpha1.CASConfigKey)]
 
-	// get the storage class name corresponding to this volume
+	cloneLabels, err := v.getCloneLabels()
```

Comment:
Why should a clone related invocation be done from within a volume creation? Can we preserve the volume operations in this file isolated to volume only? In other words can we avoid  if else logic in volume code to accommodate clone?

Reply:
This is a volume creation call. The only difference is the data source. I can't think of any other place to keep this. The caller of `Create` is `create` which performs initial validation, putting this logic there may not be any beneficial.

Reply:
`volumeSpecificRequest` -> `volumeV1alpha1SpecificRequest` -> `Create` -> `create`